builds/e2e/nested-e2e.yaml

```diff
@@ -40,11 +40,19 @@ resources:
     branch: 'release/1.2'
 
 stages:
-- stage: SetupVM
+- stage: LockAgents 
+  jobs:
+  - template: templates/lock-test-agents.yaml
+    parameters:
+      upstream.protocol: 'mqtt'
+
+- stage: RunNestedTests
+  dependsOn: LockAgents
   jobs:
   - template: templates/nested-parent-vm-setup.yaml
     parameters:
       upstream.protocol: mqtt
+      test.l4DeploymentFileName: 'nestededge_middleLayerBaseDeployment_mqtt.json'
```

Comment:
We don't need any special deployment for the mqtt upstream middle level, as we don't use any test modules on this level.